### PR TITLE
Use 'tr -d' instead of 'base64 -w0'

### PR DESCRIPTION
**Pull request by roberthbailey**
**What this PR does / why we need it**: To be compatible with running the scripts on macOS.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

@kubernetes/kube-deploy-reviewers

### CODE REVIEW

**Comment by roberthbailey** [comment]
For some background about why we do it this way, see https://github.com/kubernetes/kubernetes/pull/9543#issuecomment-110588736

**Comment by ingvagabund** [comment]
How hard would it be to support the Mac OS part with CI tests? Given the change are in `examples` only, sooner or later users will be able to detect if it breaks. I assume it is not a general practice to have CI for anything that falls under examples, right?

File: clusterctl/examples/vsphere/generate-yaml.sh [hunk 1]
```diff
@@ -53,11 +53,11 @@ fi
 
 OS=$(uname)
 if [[ "$OS" =~ "Linux" ]]; then
-  MACHINE_CONTROLLER_SSH_PUBLIC=$(cat $MACHINE_CONTROLLER_SSH_HOME$MACHINE_CONTROLLER_SSH_PUBLIC_FILE|base64 -w0)
-  MACHINE_CONTROLLER_SSH_PRIVATE=$(cat $MACHINE_CONTROLLER_SSH_HOME$MACHINE_CONTROLLER_SSH_PRIVATE_FILE|base64 -w0)
+  MACHINE_CONTROLLER_SSH_PUBLIC=$(cat $MACHINE_CONTROLLER_SSH_HOME$MACHINE_CONTROLLER_SSH_PUBLIC_FILE | base64 | tr -d '\r\n')
```
Comment (k4leung4) [inline clusterctl/examples/vsphere/generate-yaml.sh]: The original reason to introduce the OS check here is to handle the lack of -w0 option on Mac.
Seems like by using "tr -d '\r\n'", we can get rid of the OS check completely as the Linux codepath is the same as the Mac codepath.

Comment (roberthbailey) [reply clusterctl/examples/vsphere/generate-yaml.sh]: Good point. I just did a search/replace, but collapsing to a single set of commands that work for both is definitely better.

**Comment by roberthbailey** [comment]
Re-pushed.

**Comment by roberthbailey** [comment]
@ingvagabund - AFAIK we never added support for the mac part with CI for the kubernetes/kubernetes repo when we made similar changes (and I don't know if there is a good CI environment that will run on a Mac).

**Comment by k4leung4** [comment]
lgtm

File: clusterctl/examples/google/generate-yaml.sh [hunk 2]
```diff
@@ -37,7 +37,7 @@ MACHINE_CONTROLLER_SSH_PUBLIC=
 MACHINE_CONTROLLER_SSH_PRIVATE_FILE=${OUTPUT_DIR}/machine-controller-key
 MACHINE_CONTROLLER_SSH_PRIVATE=
 MACHINE_CONTROLLER_SSH_USER_PLAIN=clusterapi
-MACHINE_CONTROLLER_SSH_USER=$(echo -n "$MACHINE_CONTROLLER_SSH_USER_PLAIN" | base64 -w0)
+MACHINE_CONTROLLER_SSH_USER=$(echo -n "$MACHINE_CONTROLLER_SSH_USER_PLAIN" | base64 | tr -d '\r\n')
```
Comment (k4leung4) [inline clusterctl/examples/google/generate-yaml.sh]: nit: can you document the reason for tr -d similar to what you did in the other PR.

**Comment by spew** [comment]
lgtm once kenny's last comment is addressed (either in or out I have feelings both ways on both documenting and not documenting)

**Comment by roberthbailey** [comment]
I've added comments to each usage of `base64`. I think we should shoot to replace this with real code "soon" so I don't think it's worth making the bash work on windows at this point.

**Pull request merged**